shared/pingpong/accounts.go

```diff
@@ -667,28 +667,37 @@ func (pps *WorkerState) prepareApps(client *libgoal.Client) (err error) {
 	}
 
 	// generate new apps
+	// cycle through accts and create apps until the desired quantity is reached
 	var txgroup []transactions.Transaction
 	var senders []string
-	for addr, acct := range pps.accounts {
-		if len(pps.cinfo.AppParams) >= int(pps.cfg.NumApp) {
-			break
-		}
-		var tx transactions.Transaction
-		tx, err = pps.newApp(addr, client)
-		if err != nil {
-			return
-		}
-		acct.addBalance(-int64(pps.cfg.MaxFee))
-		txgroup = append(txgroup, tx)
-		senders = append(senders, addr)
-		if len(txgroup) == int(pps.cfg.GroupSize) {
-			pps.schedule(len(txgroup))
-			err = pps.sendAsGroup(txgroup, client, senders)
+	appsPerAddr := make(map[string]int)
+	totalAppCnt := 0
```

Comment:
This loses the possibility that accounts with apps already exist. One of the big changes in my recent pingpong hacking was that we can now have stable accounts and apps and assets that exist over multiple invocations of pingpong. Maybe `existingAppCnt := len(pps.cinfo.AppParams)` and then loop while `newAppCnt + existingAppCnt < pps.cfg.NumApp`

Reply:
Ah got it. That explains a few things I was confused about, like why some failing pingpong invocations worked on subsequent runs. Your suggestion makes sense.